### PR TITLE
[quest] Fix 'Blood Theory' Objectives

### LINKED ISSUE #6175: Quest Issue for quest 14304 - Blood Theory (Desolace, Cataclysm Classic)
## Description
For this quest you have to throw an item at a Rejuvenated Thunder Lizard. After then killing the lizard, a blood filled leech will spawn near the body that you can pick up. Neither the lizard nor the leech shows up as mobs of interest for the quest, or show up on the map where to find them.

## Wowhead link
Quest: https://www.wowhead.com/cata/quest=14304/blood-theory
Lizard: https://www.wowhead.com/cata/npc=35412/rejuvenated-thunder-lizard
Leech: https://www.wowhead.com/cata/item=49136/blood-filled-leech#created-by-spell

## Questie & WoW version
Questie v10.3.5 & Cataclysm Classic

### PR DESCRIPTION
**Pull request by Sommos**
## Issue references

Fixes #6175

## Proposed changes

- Add a fix to `objectives` for the 'Blood Theory' quest.

### CODE REVIEW

File: Database/Corrections/cataQuestFixes.lua [hunk 1]
```diff
@@ -1644,6 +1644,9 @@ function CataQuestFixes.Load()
             [questKeys.objectives] = {},
             [questKeys.triggerEnd] = {"Fire Trial Completed",{[zoneIDs.AZSHARA] = {{33.43,23.39}}}},
         },
+        [14304] = { -- Blood Theory
+            [questKeys.objectives] = {{{25412,nil,Questie.ICON_TYPE_INTERACT},nil,{{49136}},nil}},
```
Comment (BreakBB) [inline Database/Corrections/cataQuestFixes.lua:1648]: This does not work.

It is important, that the count of `objectives` in the Questie DB matches, the actual amount of objectives given by `/dump C_QuestLog.GetQuestObjectives(14304)`. The API says there is only a single objective of type "item". And that is already correct in the DB.

What we need instead is an `extraObjective` to also point to NPC 25412

Comment (Sommos) [reply Database/Corrections/cataQuestFixes.lua]: https://github.com/Questie/Questie/pull/6179/commits/07b43e9d57f77ce52ec0ca9f4297bb7158022a50

File: Database/Corrections/cataQuestFixes.lua [hunk 2]
```diff
@@ -1644,6 +1644,9 @@ function CataQuestFixes.Load()
             [questKeys.objectives] = {},
             [questKeys.triggerEnd] = {"Fire Trial Completed",{[zoneIDs.AZSHARA] = {{33.43,23.39}}}},
         },
+        [14304] = { -- Blood Theory
+            [questKeys.extraObjectives] = {{nil, Questie.ICON_TYPE_INTERACT, l10n("Collect 10 Blood-filled Leeches"), 0, {{"monster", 25412}}}},
```
Comment (BreakBB) [inline Database/Corrections/cataQuestFixes.lua:1648]: When adding a new string that is handed to `l10n()` then that string also needed to be added to the translations.

The extraObjective strings are stored here: https://github.com/Questie/Questie/tree/master/Localization/Translations/ExtraObjectives

Comment (Sommos) [reply Database/Corrections/cataQuestFixes.lua:1648]: https://github.com/Questie/Questie/pull/6179/commits/28b6d2741f66c25cc2d4a83800c89fe404d98a47

**Review by BreakBB — APPROVED** [review]

File: Database/Corrections/cataQuestFixes.lua [hunk 3]
```diff
@@ -1665,6 +1665,9 @@ function CataQuestFixes.Load()
...
```
Comment (BreakBB) [inline Database/Corrections/cataQuestFixes.lua:1669]:
```suggestion
            [questKeys.extraObjectives] = {{nil, Questie.ICON_TYPE_INTERACT, l10n("Throw Bottle of Leeches"), 0, {{"monster", 35412}}}},
```

File: Localization/Translations/ExtraObjectives/CataObjectives.lua [hunk 4]
```diff
@@ -2594,6 +2594,18 @@ local cataObjectiveLocales = {
         ["zhTW"] = false,
         ["zhCN"] = false,
     },
+    ["Collect 10 Blood-filled Leeches"] = {
```
Comment (BreakBB) [inline Localization/Translations/ExtraObjectives/CataObjectives.lua:2597]:
```suggestion
    ["Throw Bottle of Leeches"] = {
```

**Pull request merged**